$NetBSD$

The python generated header files somehow manage to override the
_NETBSD_SOURCE macro and causes build failures for tests.

Temporary fix is to pass -D_NETBSD_SOURCE as a compiler flag.

XXX: Investigate this issue futher and see why it is causing
build failures.

--- CMakeLists.txt.orig	2019-05-06 01:35:03.789287492 +0000
+++ CMakeLists.txt
@@ -49,9 +49,9 @@ if(MSVC)
 		set(STRICT_OPTIONS_CPP "${STRICT_OPTIONS_CPP} /WX")
 	endif()
 else()
-        set(STRICT_OPTIONS_CXX "${STRICT_OPTIONS_CXX}  -std=c++11 -O2 ")
-	set(STRICT_OPTIONS_CPP "${STRICT_OPTIONS_CPP}  -Wall -Wuninitialized -Wno-deprecated-declarations -Wno-missing-field-initializers ")
-	set(STRICT_OPTIONS_C "${STRICT_OPTIONS_C}  -std=c99 -O2 -Wstrict-prototypes -Wno-error=strict-prototypes -ffunction-sections -fdata-sections -fvisibility=hidden -fomit-frame-pointer ")
+        set(STRICT_OPTIONS_CXX "${STRICT_OPTIONS_CXX} -D_NETBSD_SOURCE -std=c++11 -O2 ")
+	set(STRICT_OPTIONS_CPP "${STRICT_OPTIONS_CPP} -D_NETBSD_SOURCE -Wall -Wuninitialized -Wno-deprecated-declarations -Wno-missing-field-initializers ")
+	set(STRICT_OPTIONS_C "${STRICT_OPTIONS_C}  -D_NETBSD_SOURCE -std=c99 -O2 -Wstrict-prototypes -Wno-error=strict-prototypes -ffunction-sections -fdata-sections -fvisibility=hidden -fomit-frame-pointer ")
 	if(CMAKE_C_COMPILER_ID STREQUAL "Clang")
 		set(STRICT_OPTIONS_CPP "${STRICT_OPTIONS_CPP} -Qunused-arguments -Wno-array-bounds ")
 	endif()
